azure-iot-device/tests/iothub/aio/test_async_clients.py

```diff
@@ -1049,6 +1065,370 @@ class TestIoTHubDeviceClientCreateFromSharedAccessSignature(
     pass
 
 
+@pytest.mark.describe(
+    "IoTHubDeviceClient (Synchronous) - .create_from_registration_result() -- Error"
+)
+class TestIoTHubDeviceClientCreateFromRegistrationResultError(IoTHubDeviceClientTestsConfig):
```

Comment:
>TestIoTHubDeviceClientCreateFromRegistrationResultError [](start = 6, length = 55)

Having a single implementation that is shared between sync and async makes me wonder if we should have a single set of tests that are shared between sync and async.  It looks like `TestIoTHubDeviceClientCreateFromSharedAccessSignature`  and `TestIoTHubDeviceClientCreateFromConnectionString` do this, maybe `TestIoTHubDeviceClientCreateFromRegistrationResult*` should too.